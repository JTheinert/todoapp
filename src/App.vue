<template>
  <div
    id="app"
    class="container">
    <div class="header">
      <img
        src="./components/assets/images.png"
        class="logo" >

      <h1> Vue to Do List!</h1>
    </div>
    <router-view
      default="{name: 'home'}"/>
</div></template>

<script>
import Overview from './components/Overview'
import Details from './components/Details'
import Delete from './components/Delete'
import Testing from './components/testing'

export default {
  nme: 'app',
  components: {

    Overview,
    Details,
    Delete,
    Testing
  }
}
</script>

<style>
.h1 {
font-family: "Rockwell Extr Bold", "Rockwell Bold", monospace;
font-size: 24px;
font-style: italic;
font-variant: normal;
font-weight: 700;
line-height: 26.4px;
}
* {
box-sizing: border-box;
}
.container {
max-width: 100%;
margin-left: 100px;
margin-right: 100px;
margin-top: 50px;
}
#app {
font-family: "Rockwell Extr Bold", "Rockwell Bold", monospace;
-webkit-font-smoothing: antialiased;
-moz-osx-font-smoothing: grayscale;
color: #2c3e50;
margin-top: 30px;
font-size: 26px;
}
.logo {
margin: 20px;
height: 90px;
margin-top: 10px;
}
.header {
display: flex;
justify-content: center;
}
</style>
