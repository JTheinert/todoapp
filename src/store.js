/* eslint-disable no-console */
import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export const store = new Vuex.Store({
  mutations: {
    updateIsOpen (state, { id, isOpen }) {
      const todo = state.todos.find(obj => obj.id === id)
      todo.isOpen = isOpen
    },
    updateFilteredNr () {
      // eslint-disable-next-line no-use-before-define
      const filteredNr = filteredNr
    },

    set (newTodo) {
      return newTodo
    },
    updateNummer (state, { nummer }) {
      state.nummer = nummer
    },
    updateIdForTodo (state) {
      state.idForTodo++
    }
  },
  methods: {
    mounted () {
      // eslint-disable-next-line no-console
      console.log('App mounted!')
      if (localStorage.getItem('todos')) { this.todos = JSON.parse(localStorage.getItem('todos')) }
    },
    watch: {
      todos: {
        handler () {
          console.log('Todos changed!')
          localStorage.setItem('todos', JSON.stringify(this.todos))
        },
        deep: true
      }
    }
  },
  state: {
    filteredNr: 1,
    modalShow: false,
    newTodo: '',
    idForTodo: 1,
    beforeEditCache: '',
    beforeEditDescriptionCache: '',
    placeholder: 'Edit Description',
    dialog: false,
    nummer: 0,
    todos: [
      {
        id: 0,
        title: 'cool',
        editing: false,
        descriptionEditing: false,
        isOpen: false,
        status: "isn't done",
        description: '',
        filter: false,
        filterNr: 1,
        used: 3
      }
    ]
  },

  getters: {
    doneTodos: state => {
      return state.todos.filter(todo => todo.done)
    },
    doneTodosCount: (getters) => {
      return getters.doneTodos.length
    },
    getTodoById: state => id => {
      return state.todos.find(todo => todo.id === id)
    },
    getName: () => 'rajesh'
  }
})
