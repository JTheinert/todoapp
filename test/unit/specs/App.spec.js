import App from '../../../src/App.vue'
import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import { store } from '@/store'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('App.vue', () => {
  test('Test', () => {
    expect(true).toBe(true)
  })
  test('App has a Header', () => {
    const wrapper = mount(App, {store, VueRouter, localVue})
    expect(wrapper.text()).toContain('A Vue to Do List!')
  })
  test('renders correctly', () => {
    const wrapper = mount(App)
    expect(wrapper.element).toMatchSnapshot()
  })
})
