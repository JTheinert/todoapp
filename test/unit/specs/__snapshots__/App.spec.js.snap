// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders correctly 1`] = `
<div
  class="container"
  id="app"
>
  <div
    class="header"
  >
    <img
      class="logo"
      src="./components/assets/images.png"
    />
     
    <h1>
      A Vue to Do List!
    </h1>
  </div>
   
  <router-view
    default="{name: 'home'}"
  />
</div>
`;
